homeassistant/components/envisalink/alarm_control_panel.py

```diff
@@ -173,6 +176,17 @@ def state(self):
         """Alarm trigger command. Will be used to trigger a panic alarm."""
         self.hass.data[DATA_EVL].panic_alarm(self._panic_type)
 
+    async def async_alarm_arm_night(self, code=None):
+        """Send arm night command."""
+        if code:
+            self.hass.data[DATA_EVL].arm_night_partition(
+                str(code), self._partition_number
+            )
+        else:
+            self.hass.data[DATA_EVL].arm_night_partition(
+                str(self._code), self._partition_number
+            )
```

Comment:
```suggestion
    self.hass.data[DATA_EVL].arm_night_partition(
        code if code is not None else str(self._code), self._partition_number
    )
```

I didn't check the pyenvisalink thoroughly, but it looks like its `send_data` function is not an async call, this usage may cause I/O operation in event loop. We probably need re-examine this integration

Reply:
Happy to look at converting pyenvisalink's send_data to an async call after this PR is merged